### PR TITLE
Profile Restructuring

### PR DESCRIPTION
**Pull request by shawndwells**
Adjusts the RHEL7 and RHEL8 OSPP profiles to have the same structure. Makes it easier to see what is missing between them.

Missing rules were added via comments/placeholders for future discussion. For examples see the RHEL7 audit section.

### CODE REVIEW

File: rhel7/profiles/ospp.profile [hunk 1]
```diff
@@ -83,104 +90,454 @@ selections:
...
-    - sshd_disable_kerb_auth
-    - sshd_disable_gssapi_auth
-    - sshd_idle_timeout_value=10_minutes
-    - sshd_set_keepalive
-    - sshd_enable_warning_banner
-    - sshd_disable_rhosts_rsa
-    - sshd_use_approved_ciphers
-    - sshd_use_approved_macs
+    ### Module Blacklist
+    #- kernel_module_usb-storage_disabled
```
Comment (dahaic) [inline rhel7/profiles/ospp.profile]: Ahoj Shawn. This is one rule that you missed - it was enabled before. Can you uncomment it?

File: rhel8/profiles/ospp.profile [hunk 2]
```diff
@@ -5,1220 +5,540 @@ title: 'Protection Profile for General Purpose Operating Systems'
...
+    ## FMT_MOF_EXT.1
+    - package_tmux_installed
+    - configure_tmux_lock_command
 
-    ## Record Unsuccessful Creation Attempts to Files - open O_CREAT
-    - audit_rules_unsuccessful_file_modification_open_o_creat
+    ## Set Screen Lock Timeout Period to 30 Minutes or Less
+    ## AC-11(a) / FMT_MOF_EXT.1
+    #- sshd_idle_timeout_value=10_minutes
+    #- sshd_set_idle_timeout
```
Comment (dahaic) [inline rhel8/profiles/ospp.profile]: Should be uncommented

File: rhel8/profiles/ospp.profile [hunk 3]
```diff
@@ -5,1220 +5,540 @@ title: 'Protection Profile for General Purpose Operating Systems'
...
-    ## Misc Partitioning
-    ###########
-    ## Add nodev Option to Non-Root Local Partitions
-    - mount_option_nodev_nonroot_local_partitions
+    ### Module Blacklist
+    - kernel_module_usb-storage_disabled
+    - kernel_module_cramfs_disabled
+    - kernel_module_bluetooth_disabled
+    #- kernel_module_dccp_disabled
+    #- kernel_module_sctp_disabled
```
Comment (dahaic) [inline rhel8/profiles/ospp.profile]: Should be uncommented

File: rhel8/profiles/ospp.profile [hunk 4]
```diff
@@ -5,1220 +5,540 @@ title: 'Protection Profile for General Purpose Operating Systems'
...
+    - kernel_module_firewire-core_disabled
+    - kernel_module_atm_disabled
+    - kernel_module_can_disabled
+    - kernel_module_tipc_disabled
 
-    #################################################################
-    ## Required Packages
-    #################################################################
+    ### rpcbind
+    - service_rpcbind_disabled
```
Comment (dahaic) [inline rhel8/profiles/ospp.profile]: Should be commented out (has not been present).

File: rhel8/profiles/ospp.profile [hunk 5]
```diff
@@ -5,1220 +5,540 @@ title: 'Protection Profile for General Purpose Operating Systems'
...
+    - audit_rules_session_events
+    - audit_rules_mac_modification
 
-    ###########
-    ## Misc Partitioning
-    ###########
-    ## Add nodev Option to Non-Root Local Partitions
-    - mount_option_nodev_nonroot_local_partitions
+    ### Module Blacklist
+    - kernel_module_usb-storage_disabled
```
Comment (dahaic) [inline rhel8/profiles/ospp.profile]: Should be commented out

**Comment by dahaic** [comment]
@yuumasato can you take a look - do we want to preserve "TODO" lines in the profile, or is this information superfluous?

File: rhel8/profiles/ospp.profile (hunk 3 above)
Comment (yuumasato) [reply rhel8/profiles/ospp.profile]: Uncomment only `kernel_module_sctp_disabled`.
And remove `kernel_module_dccp_disabled`, this one is part of rhel7 only, not rhel8.

File: rhel8/profiles/ospp.profile (hunk 5 above)
Comment (yuumasato) [reply rhel8/profiles/ospp.profile]: This is on rhel7 config only.

File: rhel8/profiles/ospp.profile (hunk 4 above)
Comment (yuumasato) [reply rhel8/profiles/ospp.profile]: This is on rhel7 config only.

File: rhel8/profiles/ospp.profile [hunk 6]
```diff
@@ -5,1220 +5,540 @@ title: 'Protection Profile for General Purpose Operating Systems'
...
     - dnf-automatic_security_updates_only
 
-    ## Configure dnf-automatic to Install Available Updates Automatically
+    # Configure dnf-automatic to Install Available Updates Automatically
     - dnf-automatic_apply_updates
 
-    ## Enable dnf-automatic Timer
-    - timer_dnf-automatic_enabled
+    # Enable dnf-automatic Timer
+    - timer_dnf-automatic_enabled
```
Comment (yuumasato) [inline rhel8/profiles/ospp.profile]: Make sure there is a new line at end of file.

File: rhel7/profiles/ospp.profile [hunk 7]
```diff
@@ -83,104 +90,454 @@ selections:
...
+    ## Enable Automatic Software Updates
+    ## SI-2 / FMT_MOF_EXT.1
+    # Configure dnf-automatic to Install Only Security Updates
+    #- dnf-automatic_security_updates_only
+
+    # Configure dnf-automatic to Install Available Updates Automatically
+    #- dnf-automatic_apply_updates
+
+    # Enable dnf-automatic Timer
+    #- timer_dnf-automatic_enabled
```
Comment (yuumasato) [inline rhel7/profiles/ospp.profile]: Make sure there is a new line at end of file.

File: rhel7/profiles/ospp.profile (hunk 1 above)
Comment (shawndwells) [reply rhel7/profiles/ospp.profile]: Done, and thanks for the rule review. A surprising amount of selections to track.

File: rhel8/profiles/ospp.profile (hunk 2 above)
Comment (shawndwells) [reply rhel8/profiles/ospp.profile]: Done.

File: rhel8/profiles/ospp.profile (hunk 3 above)
Comment (shawndwells) [reply rhel8/profiles/ospp.profile]: Uncommented ``kernel_module_sctp_disabled``.

Where can we find docs saying that ``dccp`` is no longer a possible kernel module on RHEL 8? Should we leave commented for now, to merge PR, and remove once docs are found?

File: rhel8/profiles/ospp.profile (hunk 4 above)
Comment (shawndwells) [reply rhel8/profiles/ospp.profile]: Updated.

File: rhel8/profiles/ospp.profile (hunk 5 above)
Comment (shawndwells) [reply rhel8/profiles/ospp.profile]: Updated.

File: rhel8/profiles/ospp.profile (hunk 6 above)
Comment (shawndwells) [reply rhel8/profiles/ospp.profile]: Updated both RHEL7 and RHEL8 profiles with new lines.

File: rhel8/profiles/ospp.profile (hunk 3 above)
Comment (yuumasato) [reply rhel8/profiles/ospp.profile]: Couldn't find documentation about it.
On RHEL8, `dccp` module is not found, `xt_dccp` is there though:
```
# find /lib/modules/ -name *dccp*
/lib/modules/4.18.0-80.el8.x86_64/kernel/net/netfilter/xt_dccp.ko.xz
```
Don't know if `xt_dccp` is relevant for this rule.

I'm okay leaving it commented for now.

**Comment by yuumasato** [comment]
@shawndwells Can you rebase and fix confilcts? Thanks

**Comment by yuumasato** [comment]
@shawndwells Also, can you clarify what is the difference between:

```
    ### GENERAL REQUIREMENTS
    ### Things needed to meet OSPP functional requirements.
```

and 

```
    ### CONFIGURATION ANNEX TO THE PROTECTION PROFILE
    ### FOR GENERAL PURPOSE OPERATING SYSTEMS
    ### ANNEX RELEASE 1
    ### FOR PROTECTION PROFILE VERSIONS 4.2
    ###
    ### https://www.niap-ccevs.org/MMO/PP/-442ConfigAnnex-/
```

**Comment by matejak** [comment]
I have a remark concerning layout (i.e. not selections): The PR changes the RHEL8 profile file quite a lot. This is undesirable for the following reasons:

- The RHEL8 profile covers more rules, so it makes sense to try to keep the changed lines of the big profile to a minimum, and change the RHEL7 profile accordingly. Ideally, the RHEL7 and RHEL8 profile files should have a similar structure, so they are easy to compare.
- RHEL8 content is under heavy development, respective PRs are opened, and modifying the profile file heavily means many merge conflicts.

**Comment by yuumasato** [comment]
> I have a remark concerning layout (i.e. not selections): The PR changes the RHEL8 profile file quite a lot. This is undesirable for the following reasons:
> 
>     * The RHEL8 profile covers more rules, so it makes sense to try to keep the changed lines of the big profile to a minimum, and change the RHEL7 profile accordingly. Ideally, the RHEL7 and RHEL8 profile files should have a similar structure, so they are easy to compare.
> 
>     * RHEL8 content is under heavy development, respective PRs are opened, and modifying the profile file heavily means many merge conflicts.

I understand your concerns.
I apologize for disregarding other PRs and conflicts that would be created.

**Comment by shawndwells** [comment]
> I have a remark concerning layout (i.e. not selections): The PR changes the RHEL8 profile file quite a lot. This is undesirable for the following reasons:
> 
> * The RHEL8 profile covers more rules, so it makes sense to try to keep the changed lines of the big profile to a minimum, and change the RHEL7 profile accordingly. Ideally, the RHEL7 and RHEL8 profile files should have a similar structure, so they are easy to compare.
> * RHEL8 content is under heavy development, respective PRs are opened, and modifying the profile file heavily means many merge conflicts.

They'll have to rebase.

**Comment by shawndwells** [comment]
> > I have a remark concerning layout (i.e. not selections): The PR changes the RHEL8 profile file quite a lot. This is undesirable for the following reasons:
> > 
> > * The RHEL8 profile covers more rules, so it makes sense to try to keep the changed lines of the big profile to a minimum, and change the RHEL7 profile accordingly. Ideally, the RHEL7 and RHEL8 profile files should have a similar structure, so they are easy to compare.
> > * RHEL8 content is under heavy development, respective PRs are opened, and modifying the profile file heavily means many merge conflicts.
> 
> They'll have to rebase.

e.g. by that same logic, those other PRs are causing merge conflicts here.

**Comment by matejak** [comment]
test this please

**Comment by matejak** [comment]
CI has found issues with some selections in the RHEL7 being there twice. Next, `grub2_enable_fips_mode` fell out of RHEL7 OSPP (is it intentional?), and `service_rngd_enabled` should be deselected in the RHEL8 profile.

Another rebase is needed, though.

**Comment by yuumasato** [comment]
> CI has found issues with some selections in the RHEL7 being there twice.

Namely:
```
Duplicate key-sequence ['ensure_redhat_gpgkey_installed'] in unique identity-constraint '{http://checklists.nist.gov/xccdf/1.1}itemSelectKey'.
Duplicate key-sequence ['ensure_gpgcheck_globally_activated'] in unique identity-constraint '{http://checklists.nist.gov/xccdf/1.1}itemSelectKey'.
Duplicate key-sequence ['ensure_gpgcheck_local_packages'] in unique identity-constraint '{http://checklists.nist.gov/xccdf/1.1}itemSelectKey'.
Duplicate key-sequence ['ensure_gpgcheck_never_disabled'] in unique identity-constraint '{http://checklists.nist.gov/xccdf/1.1}itemSelectKey'.
```

> Next, `grub2_enable_fips_mode` fell out of RHEL7 OSPP (is it intentional?)

The rule should stay.

> and `service_rngd_enabled` should be deselected in the RHEL8 profile.

Why? It was part of the profile, and is part of the target configuration.

@shawndwells Please, also mind the audit rules, that for now are consolidated into `audit_rules_for_ospp`.

**Comment by adelton** [comment]
> > and `service_rngd_enabled` should be deselected in the RHEL8 profile.
> 
> Why? It was part of the profile, and is part of the target configuration.

I assume it's a type and @matejak meant "selected", not "deselected".

**Comment by matejak** [comment]
Yes, my mistake, I have confused the `service_rngd_enabled` status, it is supposed to be enabled.

**Comment by matejak** [comment]
This PR is superseded by #4754, which basically evolves this PR and fixes conflicts and CI failures.

**Pull request closed without merging**